VibeSomeMauiBro.UITests/BaseTest.cs

```diff
@@ -0,0 +1,154 @@
+using OpenQA.Selenium.Appium;
+using OpenQA.Selenium.Appium.Android;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace VibeSomeMauiBro.UITests;
+
+public abstract class BaseTest : IDisposable
+{
+    protected AndroidDriver Driver { get; private set; } = null!;
+    
+    public string PackageName { get; } = "com.companyname.vibesomemauibro";
+    public string ActivityName { get; } = "com.companyname.vibesomemauibro.MainActivity";
+    
+    private static readonly string ArtifactsPath = GetArtifactsPath();
+    
+    private static string GetArtifactsPath()
+    {
+        string? assemblyLocation = Assembly.GetExecutingAssembly().Location;
+        if (string.IsNullOrEmpty(assemblyLocation))
+        {
+            Console.WriteLine("Warning: Assembly location is null or empty, falling back to current directory");
+            return Path.Combine(Environment.CurrentDirectory, "test-artifacts");
+        }
+
+        string? assemblyDirectory = Path.GetDirectoryName(assemblyLocation);
+        if (string.IsNullOrEmpty(assemblyDirectory))
+        {
+            Console.WriteLine("Warning: Assembly directory is null or empty, falling back to current directory");
+            return Path.Combine(Environment.CurrentDirectory, "test-artifacts");
+        }
+
+        // Navigate up from bin/Debug/net9.0/ to project root, then create test-artifacts
+        string artifactsPath = Path.Combine(assemblyDirectory, "..", "..", "..", "test-artifacts");
+        string fullPath = Path.GetFullPath(artifactsPath);
+        
+        Console.WriteLine($"Assembly location: {assemblyLocation}");
+        Console.WriteLine($"Assembly directory: {assemblyDirectory}");
+        Console.WriteLine($"Artifacts path: {fullPath}");
+        
+        return fullPath;
+    }
+    
+    static BaseTest()
+    {
+        // Ensure artifacts directory exists
+        Directory.CreateDirectory(ArtifactsPath);
+        Console.WriteLine($"Created artifacts directory: {ArtifactsPath}");
+    }
+
+    protected void InitializeAndroidDriver()
+    {
+        var options = new AppiumOptions();
+        
+        // Basic Android capabilities
+        options.AddAdditionalAppiumOption("platformName", "Android");
+        options.AutomationName = "UiAutomator2";
+        options.AddAdditionalAppiumOption("appPackage", PackageName);
+        options.AddAdditionalAppiumOption("appActivity", ActivityName);
+        options.AddAdditionalAppiumOption("appium:newCommandTimeout", 300);
+        options.AddAdditionalAppiumOption("appium:connectHardwareKeyboard", true);
+
+        // Create driver with default Appium server URL
+        var serverUri = new Uri("http://127.0.0.1:4723");
+        Driver = new AndroidDriver(serverUri, options);
+        
+        // Activate the app to ensure it's launched
+        Driver.ActivateApp(PackageName);
+    }
+
+    protected void CaptureTestFailureDiagnostics(string testName)
```

Comment:
Can `testName` use `[CallerMemberName]`:

https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute?view=net-9.0

Set this up, so none of the callers have to pass in a test name.